### PR TITLE
WIP: M10 support

### PR DESCRIPTION
**Pull request by TheSkorm**
Following on from #96 Once testing is deemed ok, this will add M10 support to autorx. This work was completed by @Viproz

### CODE REVIEW

**Comment by TheSkorm** [comment]
#94

File: auto_rx/build.sh [hunk 1]
```diff
@@ -15,6 +15,10 @@ gcc -c demod_dft.c
 gcc rs92dm_dft.c demod_dft.o -lm -o rs92ecc -I../ecc/ -I../rs92
 gcc rs41dm_dft.c demod_dft.o -lm -o rs41ecc -I../ecc/ -I../rs41
 gcc dfm09dm_dft.c demod_dft.o -lm -o dfm09ecc -I../ecc/ -I../dfm
+# Build M10 decoder
+cd ../m10/
+g++ M10.cpp M10Decoder.cpp M10GeneralParser.cpp M10GTopParser.cpp M10PtuParser.cpp AudioFile.cpp -lm -o m10
```
Comment (TheSkorm) [inline auto_rx/build.sh]: May require c++11 defined to build correctly.
(gcc 4.9.2 complains)

**Comment by TheSkorm** [comment]
Hi @Viproz,

I made some changes to allow this to build on OSX and older raspbian images. As I don't have the ability to test M10 sondes at the moment are you able to pull our testing branch and test to make sure I haven't broken any of your M10 functionality?

Cheers,

Michael.

**Comment by darksidelemm** [comment]
I think I might have accidentally reversed one of @Viproz 's commits somewhere along the line. With the current testing branch the M10 callsigns are not being prefixed with M10 - instead we just get callsigns like "809 2 14266" ending up uploaded, which is not going to work correctly. I haven't yet found the commit where this was reversed.

**Pull request merged**